### PR TITLE
Support for ADP8866 9-Channel LED Driver

### PR DESCRIPTION
**Pull request by psychogenic**
Added support for the ADP8866 LED driver
https://www.analog.com/media/en/technical-documentation/data-sheets/adp8866.pdf

### CODE REVIEW

File: drivers/led/adp8866.c [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+static int adp8866_led_blink(const struct device *dev, uint32_t led,
+			    uint32_t delay_on, uint32_t delay_off)
+{
+	struct adp8866_data *data = dev->data;
+	/*
+	 * TODO: FIXME -- unimplemented
+	 */
+
+	return -EIO;
+}
```
Comment (mbolivar-nordic) [inline drivers/led/adp8866.c:328]: Please consider using a "draft" PR status until it is ready to review.

**Comment by psychogenic** [comment]
Ok, fixed in last commit.

File: drivers/led/adp8866.c (hunk 1 above)
Comment (psychogenic) [reply drivers/led/adp8866.c]: optional, got rid of it and cleaned up a bit.  hopefully this is now in a satisfying condition.

**Comment by mbolivar-nordic** [comment]
Zephyr doesn't do fixup commits; please rebase your pull request and squash them together appropriately.

**Comment by psychogenic** [comment]
Ok, I do believe they are now well squashed.  This ok?

**Review by mbolivar-nordic — COMMENTED** [review]
Yes, that looks properly squashed together now. @Mani-Sadhasivam please review

**Review by Mani-Sadhasivam — CHANGES_REQUESTED** [review]
@psychogenic I haven't done the technical review of the driver but the code needs a bit of cleanup, so I've left comments for that. Please try to align this driver as like other LED drivers in the tree.

File: drivers/led/adp8866.c [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+{
+	struct adp8866_data *data = dev->data;
+
+	if (led >= APD8866_NUM_LED_DRIVERS) {
+
+		LOG_ERR("LED idx beyond max: %d", led);
+		return -EINVAL;
+	}
+
+	if (! data->setup_is_done) {
```
[the remark this answers is not in the document]
Comment (Mani-Sadhasivam) [reply drivers/led/adp8866.c:113]: If you do the setup during init, there is no need of this flag.

**Comment by aunsbjerg** [comment]
I suggest you read the [contribution guidelines](https://docs.zephyrproject.org/latest/contribute/index.html).

From a quick glance, there's a few things missing that will need fixing to comply with standards:
* signed-of-by message in the commit
* commit topic on the form: `drivers: led: add ADP8866 9-channel LED driver` (or something along those lines)
* a commit message body that explains just a little bit about the addition
* code style conformance, I suggest running clang-format, it should solve most of the issues

**Pull request closed without merging**